src/javascript/landing_pages/common.js

```diff
@@ -170,3 +174,112 @@ function setupCrowdin() {
+    const cookie_hash = {};
+    if (token_length === 32) {
+        cookie_hash.t = token.toString();
+    }
+    if (is_subsidiary) {
+        cookie_hash.s = '1';
+    }
+
+    Cookies.set('affiliate_tracking', cookie_hash, {
+        expires: 365, // expires in 365 days
+        path   : '/',
+        domain : /\.binary\.com/i.test(location.hostname) ? `.${location.hostname.split('.').slice(-2).join('.')}` : location.hostname,
+    });
+    return true;
+}
+
+function showAfffiliatePopup() {
+    const clients_country = sessionStorage.getItem('clients_country');
+    const xmlhttp = new XMLHttpRequest();
+
+    const container_id       = 'affiliate_disclaimer_popup';
+    const el_affiliate_popup = document.getElementById(container_id);
+
+    if (isLoggedIn()) return;
+    if (jpClient() || clients_country === 'jp') {
+        xmlhttp.onreadystatechange = () => {
+            if (xmlhttp.readyState === XMLHttpRequest.DONE && xmlhttp.status === 200) {
+                const div     = document.createElement('div');
+                const parser  = new DOMParser();
+                const el      = parser.parseFromString(xmlhttp.response, 'text/html');
+                const element = el.body.firstChild;
+                div.classList.add('lightbox');
+                div.appendChild(element);
+                el_affiliate_popup.appendChild(div);
+
+                const el_affiliate_btn = document.getElementById('btn_affiliate_proceed');
+                el_affiliate_btn.removeEventListener('click', close);
+                el_affiliate_btn.addEventListener('click', close);
+
+                document.body.classList.add('scroll-lock');
+            }
+        };
+
+        xmlhttp.open('GET', urlFor('affiliate_disclaimer'), true);
+        xmlhttp.send();
+    }
+
+    function close() {
+        document.body.classList.remove('scroll-lock');
+        el_affiliate_popup.remove();
+    }
+}
+
+function isLoggedIn () {
+    /**
+     * Returns the client information
+     *
+     * @param {String|null} key     The property name to return the value from, if missing returns the account object
+     * @param {String|null} loginid The account to return the value from
+     */
+    function get(key, loginid = localStorage.getItem('active_loginid')) {
+        let value;
+        const client_object = getAllAccountsObject();
+        if (key === 'loginid') {
+            value = loginid || localStorage.getItem('active_loginid');
+        } else {
+            const current_client = client_object[loginid] || getAllAccountsObject()[loginid] || {};
+
+            value = key ? current_client[key] : current_client;
+        }
+        if (!Array.isArray(value) && (+value === 1 || +value === 0 || value === 'true' || value === 'false')) {
+            value = JSON.parse(value || false);
+        }
+        return value;
+    }
+
+    function getAllAccountsObject() {
+        return JSON.parse(localStorage.getItem('client.accounts'));
```

Comment:
Maybe we should check if the item in localStorage exists first before parsing?

Reply:
it will parse and return `null` if the item doesn't exist

Reply:
Sorry, what I meant was check if it's an empty object. :p